ext/opcache/jit/zend_jit_trace.c

```diff
@@ -7915,6 +7917,15 @@ int ZEND_FASTCALL zend_jit_trace_hot_side(zend_execute_data *execute_data, uint3
 
 	if (ZEND_JIT_TRACE_NUM >= JIT_G(max_root_traces)) {
 		stop = ZEND_JIT_TRACE_STOP_TOO_MANY_TRACES;
+		/*  Traverse all JITed code, restore side exit to vm execution */
+		for (i = 1; i < ZEND_JIT_TRACE_NUM; i++) {
+			for (j = 0; j < zend_jit_traces[i].exit_count; j++) {
+				zend_jit_blacklist_trace_exit(i, j);
```

Comment:
It makes sense to avoid calling zend_jit_blacklist_trace_exit() for already jit-ed and blacklisted side exits. This would remove unnecessary locks.